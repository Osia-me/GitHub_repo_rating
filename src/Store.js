import React, { useState, useEffect } from "react";
import axios from 'axios';

export const Context = React.createContext(null);

export const Store = ({children}) => {
  const [state, setState] = useState([]);
  const linkToLastWeek = "https://api.github.com/search/repositories?q=created:%3E2021-06-13&sort=stars&order=desc";
  const linkToReposByLang = (lang) => `https://api.github.com/legacy/repos/search/language=${lang}`

  useEffect(() => {
    axios
      .get(linkToLastWeek)
      .then((response) => {
        setState({
          items: response.data.items,
          fav: [],
          view: null,
          language: 'Select language'
        });
      })
      .catch((error) => {
        alert('Something went wrong!')
      });
  }, []);

  useEffect(() => {
    if(state.language && state.language !== 'Select language'){
      axios.get(linkToReposByLang(state.language)).then((response) => {
        setState({
          items: response.data.repositories.map( el => ({
            name: el.name,
            description:el.description,
            html_url: el.url,
            stargazers_count: Math.round(el.score),
            id: Math.random() * 2.5
          })),
          fav: [],
          view: null,
          language: state.language
        });
      })
      .catch((error) => {
        alert('Something went wrong!')
      });
    }
  },[state.language])

  return (
    <Context.Provider value={[state, setState]}>
      { children }
    </Context.Provider>
  );
};

export default Store;
