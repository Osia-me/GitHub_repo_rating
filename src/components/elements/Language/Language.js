import React, { useContext } from 'react';
import { Context } from '../../../Store.js';
import "./Language.css";

function Language() {
  const [state, setState] = useContext(Context);
  const listOfLanguages = ['Javascript', 'Python', 'Go', 'Scala'];

  const languageSelectHandler = (language) => {
    setState({...state, language: language});
  }

  return (
    <nav className="nav">
        <ul className="nav__menu">
          <li className="nav__menu-item">
            <a>{state.language}</a>
            <ul className="nav__submenu">
              {listOfLanguages.map(language => (
                <li className="nav__submenu-item" key={language.length+1}>
                  <a onClick={() => languageSelectHandler(language)}>{language}</a>
                </li>
              ))}
            </ul>
          </li>
        </ul>
      </nav>
  );
}

export default Language;
