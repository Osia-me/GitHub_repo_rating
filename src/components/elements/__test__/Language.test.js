import { shallow } from 'enzyme';
import Language from '../Language/Language';
import Store from '../../../Store';

it('renders Language component', () => {
  const wrapper = shallow(
    <Store className="app">
      <Language />
    </Store>

  );
  expect(wrapper).toMatchSnapshot();
});
