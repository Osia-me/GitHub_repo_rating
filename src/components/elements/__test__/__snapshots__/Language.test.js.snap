// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Language component 1`] = `ShallowWrapper {}`;
